### ISSUE TITLE
cannot dynamic_cast class osg::ref_ptr<osg::Object>

### ISSUE THREAD

**Issue by koalatect** [OpendTect/osgGeo#1]
Hi, I'm having issues compiling this code.
Any idea ?

```
Scanning dependencies of target osgGeo
[  3%] Building CXX object src/osgGeo/CMakeFiles/osgGeo.dir/AutoTransform.cpp.o
[  7%] Building CXX object src/osgGeo/CMakeFiles/osgGeo.dir/AxesNode.cpp.o
[ 10%] Building CXX object src/osgGeo/CMakeFiles/osgGeo.dir/Callback.cpp.o
[ 14%] Building CXX object src/osgGeo/CMakeFiles/osgGeo.dir/Draggers.cpp.o
[ 17%] Building CXX object src/osgGeo/CMakeFiles/osgGeo.dir/GLInfo.cpp.o
[ 21%] Building CXX object src/osgGeo/CMakeFiles/osgGeo.dir/Palette.cpp.o
[ 25%] Building CXX object src/osgGeo/CMakeFiles/osgGeo.dir/ShaderUtility.cpp.o
[ 28%] Building CXX object src/osgGeo/CMakeFiles/osgGeo.dir/PolygonSelection.cpp.o
[ 32%] Building CXX object src/osgGeo/CMakeFiles/osgGeo.dir/PolyLine.cpp.o
[ 35%] Building CXX object src/osgGeo/CMakeFiles/osgGeo.dir/LayeredTexture.cpp.o
[ 39%] Building CXX object src/osgGeo/CMakeFiles/osgGeo.dir/LayerProcess.cpp.o
[ 42%] Building CXX object src/osgGeo/CMakeFiles/osgGeo.dir/Line3.cpp.o
[ 46%] Building CXX object src/osgGeo/CMakeFiles/osgGeo.dir/MarkerSet.cpp.o
[ 50%] Building CXX object src/osgGeo/CMakeFiles/osgGeo.dir/MarkerShape.cpp.o
[ 53%] Building CXX object src/osgGeo/CMakeFiles/osgGeo.dir/OneSideRender.cpp.o
/OpendTect-repo/osgGeo/src/osgGeo/OneSideRender.cpp: In function ‘bool readDrawables(osgDB::InputStream&, osgGeo::OneSideRender&)’:
/OpendTect-repo/osgGeo/src/osgGeo/OneSideRender.cpp:187:81: error: cannot dynamic_cast ‘osgDB::InputStream::readObject(osg::Object*)(0u)’ (of type ‘class osg::ref_ptr<osg::Object>’) to type ‘class osg::Drawable*’ (source is not a pointer)
         osg::Drawable* drawable = dynamic_cast<osg::Drawable*>( is.readObject() );
                                                                                 ^
make[2]: *** [src/osgGeo/CMakeFiles/osgGeo.dir/OneSideRender.cpp.o] Error 1
make[1]: *** [src/osgGeo/CMakeFiles/osgGeo.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by tingdahl** [comment]
Which version of osg are you using?

ons 13 juli 2016 kl 05:30 skrev koalatect notifications@github.com:

**Comment by koalatect** [comment]
Was using last version of osg.
The issue has been solved by using OpenSceneGraph-3.4.0.
Thank you!

**Comment by tingdahl** [comment]
Committed fix in master branch so it works with newest osg version. Thanks for reporting.
